genie-agent/src/main/java/com/netflix/genie/agent/execution/services/FetchingCacheService.java

```diff
@@ -0,0 +1,56 @@
+/*
+ *
+ *  Copyright 2018 Netflix, Inc.
+ *
+ *     Licensed under the Apache License, Version 2.0 (the "License");
+ *     you may not use this file except in compliance with the License.
+ *     You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *     Unless required by applicable law or agreed to in writing, software
+ *     distributed under the License is distributed on an "AS IS" BASIS,
+ *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *     See the License for the specific language governing permissions and
+ *     limitations under the License.
+ *
+ */
+
+package com.netflix.genie.agent.execution.services;
+
+import com.netflix.genie.agent.execution.exceptions.DownloadException;
+import org.apache.commons.lang3.tuple.Pair;
+
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.util.Set;
+
+/**
+ * Interface for a cache that downloads resources via URL and stores them on local disk for later reuse.
+ *
+ * @author mprimi
+ * @since 4.0.0
+ */
+public interface FetchingCacheService {
```

Comment:
It seems to me like this is more of the `DownloadService` and the other thing is more of a `ExecutionEnvironmentSetup` service or similar. it's probably just naming semantics

Reply:
Download service downloads file where you tell it to, does not know anything about environments or execution.
And this is just a very simple on-disk file cache, it does not only download but retrieve any resource (i.e. from classpath, other places on the filesystem)